docs/usage/channels.rst

```diff
@@ -0,0 +1,452 @@
+subscribers, route handlers and configuration.
+
+
+.. tip::
+    The plugin makes itself available as a dependency under the ``channels`` key, which
+    means it's not necessary to import it and instead, it can be used from within route
+    handlers or other callables within the dependency tree directly
+
+
+Configuring the channels
++++++++++++++++++++++++++
+
+The channels manged by the plugin can be either defined upfront, passing them to the
+``channels`` argument, or created "on the fly" (i.e. on the first subscription to a
+channel) by setting ``arbitrary_channels_allowed=True``.
+
+
+.. code-block:: python
+    :caption: Passing channels explicitly
+
+    from litestar.channels import ChannelsPlugin
+
+    channels_plugin = ChannelsPlugin(..., channels=["foo", "bar"])
+
+
+.. code-block:: python
+    :caption: Allowing arbitrary channels
+
+    from litestar.channels import ChannelsPlugin
+
+    channels_plugin = ChannelsPlugin(..., arbitrary_channels_allowed=True)
+
+
+If ``arbitrary_channels_allowed`` is not ``True``, trying to publish or subscribe to a
+channel not passed to ``channels`` will raise a :exc:`ChannelsException`.
+
+
+Publishing data
++++++++++++++++
+
+
+One of the core aspects of the plugin is publishing data, which is done through its
+:meth:`publish <ChannelsPlugin.publish>` method:
+
+.. code-block:: python
+
+    channels.publish({"message": "Hello"}, "general")
+
+
+The above example will publish the data to the channel ``general``, subsequently putting
+it into all subscriber's :term:`event stream` to be consumed.
+
+This method is non-blocking, even though channels and the associated
+:term:`backends <backend>` are fundamentally asynchronous.
+
+Calling ``publish`` effectively enqueues a message to be sent to the backend, from which
+follows that there's no guarantee that an event will be available in the backend
+immediately after this call.
+
+Alternatively, the asynchronous :meth:`wait_published <ChannelsPlugin.wait_published>`
+method can be used, which  the internal message queue, publishing the data to the
+backend directly.
+
+.. note::
+    While calling :meth:`publish <ChannelsPlugin.publish>` does not guarantee the
+    message is sent to the backend immediately, it will be sent there *eventually*; On
+    shutdown, the plugin will wait for all queues to empty
+
+
+Managing subscriptions
+++++++++++++++++++++++
+
+Another core functionality of the plugin is managing subscriptions, for which two
+different approaches exist:
+
+1. Manually through the :meth:`subscribe <ChannelsPlugin.subscribe>` and
+   :meth:`unsubscribe <ChannelsPlugin.unsubscribe>` methods
+2. By using the :meth:`start_subscription <ChannelsPlugin.start_subscription>` context
+   manager
+
+Both :meth:`subscribe <ChannelsPlugin.subscribe>` and
+:meth:`start_subscription <ChannelsPlugin.start_subscription>` produce a
+:class:`Subscriber`, which can be used to interact with the streams of events subscribed
```

Comment:
```suggestion
:class:`Subscriber`, which can be used to interact with the streams of events directly. 
```

Reply:
There is no method to interact with the stream indirectly, so this is a needless clarification.